modules/runner/src/integrationTest/java/org/apache/ignite/internal/runner/app/ItClusterInitTest.java

```diff
@@ -0,0 +1,138 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.runner.app;
+
+import static org.apache.ignite.internal.testframework.IgniteTestUtils.testNodeName;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasItem;
+import static org.hamcrest.Matchers.not;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import org.apache.ignite.IgnitionManager;
+import org.apache.ignite.internal.app.IgniteImpl;
+import org.apache.ignite.internal.join.Leaders;
+import org.apache.ignite.internal.testframework.WorkDirectory;
+import org.apache.ignite.internal.testframework.WorkDirectoryExtension;
+import org.apache.ignite.internal.util.IgniteUtils;
+import org.apache.ignite.lang.NodeStoppingException;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInfo;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+/**
+ * Integration tests for initializing a cluster.
+ */
+@ExtendWith(WorkDirectoryExtension.class)
+public class ItClusterInitTest {
+    @WorkDirectory
+    private Path workDir;
+
+    private final TestInfo testInfo;
+
+    private final List<IgniteImpl> nodes = new ArrayList<>();
+
+    public ItClusterInitTest(TestInfo testInfo) {
+        this.testInfo = testInfo;
+    }
+
+    @AfterEach
+    void tearDown() throws Exception {
+        IgniteUtils.closeAll(nodes);
+    }
+
+    /**
+     * Tests the happy case for cluster initialization.
+     */
+    @Test
+    void testNormalInit() throws NodeStoppingException {
+        createCluster(4);
+
+        List<String> metastorageNodes = List.of(nodes.get(0).name(), nodes.get(1).name());
+
+        List<String> cmgNodes = List.of(nodes.get(2).name(), nodes.get(3).name());
+
+        Leaders leaders = nodes.get(0).init(metastorageNodes, cmgNodes);
+
+        assertThat(metastorageNodes, hasItem(leaders.metaStorageLeader()));
+        assertThat(metastorageNodes, not(hasItem(leaders.cmgLeader())));
+
+        assertThat(cmgNodes, hasItem(leaders.cmgLeader()));
+        assertThat(cmgNodes, not(hasItem(leaders.metaStorageLeader())));
+    }
+
+    /**
+     * Tests a scenario when a cluster is initialized twice.
+     */
+    @Test
+    void testDoubleInit() throws NodeStoppingException {
+        createCluster(1);
+
+        IgniteImpl node = nodes.get(0);
+
+        node.init(List.of(node.name()));
+
+        assertThrows(Exception.class, () -> node.init(List.of(node.name())));
```

Comment:
Let's verify that the exception that is thrown is actually what we expect; otherwise, it could be anything like a `NullPointerException`